src/install.rs

```diff
@@ -74,6 +80,8 @@ impl<'a> InstallMethod<'a> {
                     prefix,
                     force_update,
                     old_date,
+                    if !exists { components } else { &[] },
```

Comment:
So this means that we'll ignore `--component` and `--target` if passed to `rustup update`? Do we also give an error for that case, or will they just be silently ignored?

Reply:
Only being for `install` was the intent, though I guess I could drop it back to allowing it for `update` too -- do you think that'd be useful?  If not, I'll add some kind of error/warning.

Reply:
No, I think keeping it just for `install` is completely reasonable. Was just wondering what would happen if I tried to pass it to `update`. If I get an error, that's excellent.

Reply:
I decided update was also useful